### DISCUSSION TITLE
Dependency injection in custom checks

### DISCUSSION THREAD

**Discussion by dennis-koster** [spatie/laravel-health#155] [Ideas]
Hi there,

Would it be possible to add the ability to use Laravel's service container for dependency injection in custom checks?

We could either add it to the `__construct`  or the `run` method of the class, to perhaps be more in line with queued jobs and their `handle` method.

Would love to hear your thoughts!

**Comment by freekmurze** [comment]
I would be open for a PR that adds this ability in a non breaking way.

**Reply by dennis-koster** [reply]
Thanks @freekmurze , I don't mind having a go at this. 

One thing I might run into, however, is that the abstract `Check` class defines the `run` method without any arguments, making it hard to inject anything there when extending that abstract class.

```php
abstract public function run(): Result;
```

The other, perhaps more logical place for this, is the constructor. But that is defined as final so can't be extended either. 

```php
final public function __construct()
{
}
```

**Reply by freekmurze** [reply]
I would be ok with make the constructor non-final

**Reply by dennis-koster** [reply]
@freekmurze Hi Freek, I took a stab at it [here](https://github.com/spatie/laravel-health/pull/157). Let me know what you think. :)
